### ISSUE TITLE
Debugger: Mismatch value between data tip and local window after editing a variable in watch window

### ISSUE THREAD

**Issue by ClaireKang** [OmniSharp/omnisharp-vscode#478]
## Environment data
`dotnet --info` output:
VS Code version:
C# Extension version:

## Steps to reproduce
1 Set breakpoint at line of var cacheKey = "topselling" in HomerController.cs
2 Press F5 to start debugging
3 Change to watch window, add expression of cacheKey = "Hello"
4 Press F10 to step over to next line, observe the data tip of  cacheKey and same variable in local window.
5 Press F10 to step over to next line

## Expected  behavior
In 4th step: The cacheKey's data tip and local window should be the same value
In 5th step: The cacheKey's data tip and local window should be "topselling",rather than "Hello";

## Actual behavior
They actually have the different values In 4th step

**Comment by gregg-miskelly** [comment]
@ClaireKang Are you removing the watch window expression after you have evaluated it once?

**Comment by ClaireKang** [comment]
No, I typed the expression in watch window while debugging and didn't remove it.

**Comment by gregg-miskelly** [comment]
@ClaireKang in that case, the behavior you are seeing seems like what I would expect -- VS Code has no smarts yet to prohibit re-evaluation of expressions that have side effects. So on each step your assignment statement will re-run at some non-deterministic point after entering break state.
